### PR TITLE
RAIL-3274 insight view title component

### PR DESCRIPTION
**Pull request by plavemar**
Description of changes.
- introduced InsightTitle component
- indroduced showTitle and TitleComponent properties for InsightView
- introduced onInsightLoaded callback property for InsightView
- added base styles for InsightTitle component

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [x] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [x] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [x] `check` passes
-   [x] `check-extended` passes
-   [x] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3397/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5040/ : FAILURE in 6m 29s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/860/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3398/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/860/ : SUCCESS in 18m 32s

**Comment by plavemar** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3399/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5041/ : SUCCESS in 15m 23s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/861/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3400/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/861/ : SUCCESS in 20m 59s

**Comment by plavemar** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR1299

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3401/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/412/ : SUCCESS in 11m 23s (non-voting)

File: libs/sdk-ui-ext/src/insightView/InsightTitle.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// (C) 2021 GoodData Corporation
+import React from "react";
+import Truncate from "react-truncate";
+/**
+ * @Public
+ */
+export interface IInsightTitle {
```
Comment (no23reason) [inline libs/sdk-ui-ext/src/insightView/InsightTitle.tsx:7]: Please use the `Props` suffix for props types, so here `IInsightTitleProps`

File: libs/sdk-ui-ext/src/insightView/InsightView.tsx [hunk 2]
```diff
@@ -163,8 +167,8 @@ class InsightViewCore extends React.Component<IInsightViewProps & WrappedCompone
         if (!insight || isEqual(insight, this.state.insight)) {
             return;
         }
-
-        this.setState({ insight });
+        this.props.onInsightLoaded?.(insight);
+        this.setState({ insight, insightTitle: this.resolveInsightTitle(insight) });
```
Comment (no23reason) [inline libs/sdk-ui-ext/src/insightView/InsightView.tsx:171]: It is not necessary to store the title in the state, right? We ca  always call the `resolveInsightTitle` whenever we want the value (the function is not heavy so there is no need to "memoize" it). And it is usually better to compute the derived data from state, not store them there.

File: libs/sdk-ui-ext/src/insightView/InsightTitle.tsx (hunk 1 above)
Comment (plavemar) [reply libs/sdk-ui-ext/src/insightView/InsightTitle.tsx:7]: Suffix addded.

File: libs/sdk-ui-ext/src/insightView/InsightView.tsx (hunk 2 above)
Comment (plavemar) [reply libs/sdk-ui-ext/src/insightView/InsightView.tsx:171]: Resolved. It seemed better to me that the title is calculated just once after insight is loaded. It is true tho that the solution without saving the title into the state has better readability.

File: libs/sdk-ui-ext/src/insightView/InsightTitle.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// (C) 2021 GoodData Corporation
+import React from "react";
+import Truncate from "react-truncate";
+/**
+ * @Public
```
Comment (no23reason) [inline libs/sdk-ui-ext/src/insightView/InsightTitle.tsx:5]: Please use lowercase in the release tags (here and the next one)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3404/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5046/ : SUCCESS in 11m 59s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1299

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1014/ : SUCCESS in 59s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1284/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5048/ : SUCCESS in 12m 09s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/415/ : SUCCESS in 9m 10s (non-voting)

**Pull request merged**